grammar.js

```diff
@@ -416,7 +423,34 @@ module.exports = grammar(C, {
       alias($.operator_cast_definition, $.function_definition),
       alias($.operator_cast_declaration, $.declaration),
       $.friend_declaration,
-      seq($.access_specifier, ':'),
+      $.access_specifier_list,
+      $.alias_declaration,
+      $.using_declaration,
+      $.type_definition,
+      $.static_assert_declaration,
+      ';',
+    ),
+
+    access_specifier_list: $ => prec.right(seq(
+      $.access_specifier,
+      ':',
+      repeat($._access_specifier_list_item),
+    )),
+
+    _access_specifier_list_item: $ => choice(
```

Comment:
I know this is resolving a particular issue, but so far I've specifically avoided doing this.

Convince me otherwise, but I don't think the visibility labels (public, private, etc), should have other field declarations as children. It introduces another potential parent for the declarations instead of them always being children of the class declaration.

I also think in cases like this, reusing the same rules (letting them support recursion or whatever) is normally a lot better from maintenance burden perspective--I'm not too excited about the duplication of nearly all field declaration rules here save the access specifier just to achieve this. 

Overall I think your PR looks good, but can we back this particular change out for now and discuss in an issue or a PR on its own?

Reply:
>  It introduces another potential parent for the declarations instead of them always being children of the class declaration.

I can understand the downsides of this - I'll remove it for tihs PR.